### ISSUE TITLE
Flameshot doesn't work on Ubuntu 22.04: Use QT_QPA_PLATFORM=wayland

### ISSUE THREAD

**Issue by Valeryan24** [flameshot-org/flameshot#2199]
### Flameshot Version

Flameshot 0.10.2

### Installation Type

Linux, MacOS, or Windows Package manager (apt, pacman, eopkg, choco, brew, ...)

### Operating System type and version

Ubuntu 22.04 development

### Description

Hi, thanks for this wonderful soft and all the work done. Sorry if the issue is linked to another one already opened, like #2107 : 

I just upgraded to Ubuntu 22.04 development branch, and Flameshot doesn't work anymore. I well have the indicator on the top Gnome Shell panel, but trying to run a screenshot does nothing - no black window with resizing borders appears.

If I try to run from terminal, I get the error message : 

`Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway`

I don't know what it means and how to set up QT_QPA_PLATFORM=wayland. I well have gambas3-gb-qt5-wayland and qtwayland5 installed, but on the login screen I only have 3 choices : Gnome, Ubuntu Wayland or X-Org.

Other users have the same issue, and the same happens with Ksnip - so it's certainly a problem due to Wayland security restrictions, and not a bug in Flameshot : 
https://forum.ubuntu-fr.org/viewtopic.php?id=2068335

### Steps to reproduce

Run flameshot on a Wayland session

### System Information

Ubuntu 22.04 on Wayland

**Comment by borgmanJeremy** [comment]
I was able to replicate this and then fix it. In a clean VM I saw this issues, then I installed: ``` xdg-desktop-portal-gnome ```, and logged out and back in and it worked fine. This package is now required per #1910 

I also installed ```bustle``` to debug any dbus issues, but i dont think this is what fixed it.

**Comment by Valeryan24** [comment]
Thanks a lot, it works fine again !

It just adds a small step: when I run the screenshot, it does a capture of the entire screen, then I have to click on "Share", and it opens the normal Flameshot border selection, with all the options to save the image :)

**Comment by borgmanJeremy** [comment]
Yes see #1910 for why that is the case, its out of our hands. @hosiet Do you think we should make these packages required dependencies in the .deb?

**Comment by mmahmoudian** [comment]
We should add this to our Wayland documentation I guess if it is reproducible on a clean VM. @borgmanJeremy and @hosiet if you two agree on this, I will assigned this to myself to do it.

**Comment by borgmanJeremy** [comment]
It for sure is a requirement for gnome 41+ on wayland. The open question is if we should list it in the .deb file also.

**Comment by hosiet** [comment]
Please do so. Also note that xdg-desktop-portal-gnome is only for GNOME. For other Wayland-enabled desktop environment, they may need xdg-desktop-portal-gtk or xdg-desktop-portal-kde or xdg-desktop-portal-wlr for different DEs respectively.

**Comment by borgmanJeremy** [comment]
@hosiet is it possible to write the control file to be specific to the DE?

**Comment by hosiet** [comment]
I don't have opinions for upstream-provides debs. For distribution-provided packages, we already have this:

https://sources.debian.org/src/flameshot/0.10.2+ds1-4/debian/control/

```
Recommends:
 xdg-desktop-portal-gtk | xdg-desktop-portal-gnome | xdg-desktop-portal-kde | xdg-desktop-portal-wlr,
```

Probably you may also use such format; it may have advantages and disadvantages. I don't have an ideal solution.

**Comment by hosiet** [comment]
Just FYI: conditional-depends (install A if B has been installed) is not supported in all known dependency resolvers since that will introduce unacceptable level of complexity.

**Comment by xvitaly** [comment]
Added to Fedora:

```
Requires: xdg-desktop-portal%{?_isa}
Requires: (xdg-desktop-portal-gnome%{?_isa} if gnome-shell%{?_isa})
Requires: (xdg-desktop-portal-kde%{?_isa} if plasma-workspace%{?_isa})
```

Should be fixed now.

**Comment by xvitaly** [comment]
> Just FYI: conditional-depends (install A if B has been installed) is not supported in all known dependency resolvers since that will introduce unacceptable level of complexity.

RPM supports this since version 4.13: https://rpm-software-management.github.io/rpm/manual/boolean_dependencies.html

**Comment by hosiet** [comment]
> RPM supports this since version 4.13: https://rpm-software-management.github.io/rpm/manual/boolean_dependencies.html

Thanks for the info. In Debian, the `gnome-core` package already has a hard dependency on `xdg-desktop-portal-gnome`. I am not sure if future Ubuntu (22.04 LTS) would also do the same.

**Comment by wgorski** [comment]
> It just adds a small step: when I run the screenshot, it does a capture of the entire screen, then I have to click on "Share", and it opens the normal Flameshot border selection, with all the options to save the image :)

Same thing here. Is this the expected behaviour?

**Comment by borgmanJeremy** [comment]
Read the link, yes, it's gnomes decision

**Comment by avramukk** [comment]
I have the same problem now. 
![image](https://user-images.githubusercontent.com/18316591/165488885-77a14256-d2b1-4cd5-bc3e-912e7bc1c7b5.png)

**Comment by mmahmoudian** [comment]
@avramukk The problem you have is with Gnome and not Flameshot. Take your complaints to them. More information here:
https://flameshot.org/docs/guide/wayland-help/#gnome

I'll hide your message as it if off-topic

**Comment by Dylan-86** [comment]
I confirm that I could make it run after installing xdg-desktop-portal-gnome and rebooting.

However, each time I take a screenshot (I've set up a custom shortcut):
- The ORIGINAL Ubuntu 22.04 screenshot utility comes up and asks me if I want to share the screenshot with the requesting app
- If I say "Share", finally Flameshot shows up.

See below... (I couldn't take a screenshot of the screenshot app :) )

![IMG_20220503_154434](https://user-images.githubusercontent.com/58813956/166563027-7efd031d-991f-44c9-8e53-22be23f13ed8.jpg)

**Comment by borgmanJeremy** [comment]
@mmahmoudian I think we should program a bot to answer that question lol.

@Dylan-86 that share button is a change gnome made upstream and we have no control over it. Sorry for the small sarcasm, I think we are averaging that question twice a day.

**Comment by mmahmoudian** [comment]
@Dylan-86 please read this:

https://flameshot.org/docs/guide/wayland-help/#gnome


@borgmanJeremy I'll pay 200€ for that bot if it answers that question every time and hide the question or close the issue. I bet Emacs has a package for that 😂

**Comment by Dylan-86** [comment]
> @Dylan-86 please read this:
> 
> https://flameshot.org/docs/guide/wayland-help/#gnome
> 
> @borgmanJeremy I'll pay 200€ for that bot if it answers that question every time and hide the question or close the issue. I bet Emacs has a package for that 😂

Thanks!!
And sorry for the complain, I got it was a gnome thing but thought that there was a workaround :( 

Thanks!

**Comment by borgmanJeremy** [comment]
I've been using gnome on xorg as a workaround but mainly use kde

**Comment by shun1249844726** [comment]
Same problem

**Comment by rachediabdenacer** [comment]
The issue with Ubuntu 22.04 is using a display feature called wayland. Wayland is not supporting to screen share & screenshot (Untill now, atleast!). In previous ubuntu version we had xorg for these features.

To check which display feature your ubuntu is using, please enter this command

```bash
echo $XDG_SESSION_TYPE
```

If the above command outputs, wayland, your system is using wayland.

We have to disable wayland and enable xorg (x11). To do this,

Please go to this file

```bash
sudo nano /etc/gdm3/custom.conf
```

Uncomment this line.

```
WaylandEnable=false
```

Reboot your system.

Voila! You can now take a screenshot. You can confirm this by re-entering this command.

```bash
echo $XDG_SESSION_TYPE
```

Now it should display x11.

**Comment by Asgoret** [comment]
Same here.  `xdg-desktop-portal-gnome` doesn't help

**Comment by VINTX2** [comment]
Same issue here.
You can set the environment variable `QT_QPA_PLATFORM=wayland` by writing it in your /etc/environment file and then reboot.
I also have xdg-desktop-portal and xdg-desktop-portal-gnome installed.

Somehow starting `flameshot gui` from the cli works.

**Comment by lgrn** [comment]
```
╰─ flameshot --version
Flameshot v12.1.0 (-)
Compiled with Qt 5.15.10
```
```
╰─ cat /etc/lsb-release
DISTRIB_ID="ManjaroLinux"
DISTRIB_RELEASE="24.0.1"
DISTRIB_CODENAME="Wynsdey"
DISTRIB_DESCRIPTION="Manjaro Linux"
```
```
╰─ sudo pacman -Q xdg-desktop-portal-gnome
xdg-desktop-portal-gnome 46.2-1
```
```
╰─ echo $XDG_SESSION_TYPE
wayland
```
Unfortunately, I still got:
```
╰─ flameshot
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
```
After googling a bit and [finding this](https://flameshot.org/docs/guide/wayland-help/), I realized that if I run `flameshot gui` it works fine, so just putting that here in case anyone else ends up here from googling.
